$OpenBSD: patch-salt_utils___init___py,v 1.4 2015/09/26 09:13:26 ajacoutot Exp $

From e697326f1b66ad032cc1febb86bf4b83b5630b6a Mon Sep 17 00:00:00 2001
From: "C. R. Oldham" <cro@ncbt.org>
Date: Fri, 11 Sep 2015 09:30:42 -0600
Subject: [PATCH] Don't check for proxy in the individual
 is_linux/is_windows/etc functions.  This breaks too many things.

--- salt/utils/__init__.py.orig	Wed Sep  9 19:00:15 2015
+++ salt/utils/__init__.py	Sat Sep 26 11:11:05 2015
@@ -1532,20 +1532,7 @@ def is_windows():
     '''
     Simple function to return if a host is Windows or not
     '''
-    import __main__ as main
-    # This is a hack.  If a proxy minion is started by other
-    # means, e.g. a custom script that creates the minion objects
-    # then this will fail.
-    is_proxy = False
-    try:
-        if 'salt-proxy' in main.__file__:
-            is_proxy = True
-    except AttributeError:
-        pass
-    if is_proxy:
-        return False
-    else:
-        return sys.platform.startswith('win')
+    return sys.platform.startswith('win')
 
 
 def sanitize_win_path_string(winpath):
@@ -1571,15 +1558,6 @@ def is_proxy():
     TODO: Need to extend this for proxies that might run on
     other Unices
     '''
-    return not (is_linux() or is_sunos() or is_windows())
-
-
-@real_memoize
-def is_linux():
-    '''
-    Simple function to return if a host is Linux or not.
-    Note for a proxy minion, we need to return something else
-    '''
     import __main__ as main
     # This is a hack.  If a proxy minion is started by other
     # means, e.g. a custom script that creates the minion objects
@@ -1590,13 +1568,19 @@ def is_linux():
             is_proxy = True
     except AttributeError:
         pass
-    if is_proxy:
-        return False
-    else:
-        return sys.platform.startswith('linux')
+    return is_proxy
 
 
 @real_memoize
+def is_linux():
+    '''
+    Simple function to return if a host is Linux or not.
+    Note for a proxy minion, we need to return something else
+    '''
+    return sys.platform.startswith('linux')
+
+
+@real_memoize
 def is_darwin():
     '''
     Simple function to return if a host is Darwin (OS X) or not
@@ -1609,20 +1593,7 @@ def is_sunos():
     '''
     Simple function to return if host is SunOS or not
     '''
-    import __main__ as main
-    # This is a hack.  If a proxy minion is started by other
-    # means, e.g. a custom script that creates the minion objects
-    # then this will fail.
-    is_proxy = False
-    try:
-        if 'salt-proxy' in main.__file__:
-            is_proxy = True
-    except AttributeError:
-        pass
-    if is_proxy:
-        return False
-    else:
-        return sys.platform.startswith('sunos')
+    return sys.platform.startswith('sunos')
 
 
 @real_memoize
